### DISCUSSION TITLE
Something like -contrast-stretch -linear-stretch -auto-level // but leaving the shadow end untouched

### DISCUSSION THREAD

**Discussion by Berrybarb** [ImageMagick/ImageMagick#2459] [Help] [answered]
Admittedly the title of the question sucks :-( What I'm looking for is any way to stretch an image to make use of full QuantumRange by a multiplication alone. 

All the operators I mention in the title seem to touch on both ends of a conceived histogram. The shadow end is tackled by a subtraction while the highlight end is by a multiplication. I need only the multiplication to take place; shadows left alone as they might fall.

Say we have a color channel and for simplicity let's talk 8-bit nomenclature 0 through 255. The darkest value is say 3 and the brightest 63. I'd need the 63 to become 255 and the 3 to become like 15. Of course depending on what is actually the maximum value found in the color channel. One operation in synch to RGB is fine, individual channel adjustments are preferred.

Can I conveniently do this with IM? Thanks.

**Comment by snibgo** [comment, marked as answer]
I think you want to divide by the maximum, on a scale of 0.0 to 1.0:
```
magick in.png -evaluate Divide %[fx:maxima] out.png

**Reply by Berrybarb** [reply]
Thank you so much. I'm so glad you made sense of what I wrote, this works perfectly.

**Comment by snibgo** [comment]
You also said:
> One operation in synch to RGB is fine, individual channel adjustments are preferred.

To adjust each channel independently (Windows BAT syntax, adjust for bash):
```
magick ^
  in.png
  -channel R -evaluate Divide %%[fx:maxima.r] ^
  -channel G -evaluate Divide %%[fx:maxima.g] ^
  -channel B -evaluate Divide %%[fx:maxima.b] ^
  +channel ^
  out.png  
```
Or this would do the same job:
```
magick ^
  in.png
  -channel RGB -separate ^
  -evaluate Divide %%[fx:maxima] ^
  -combine ^
  out.png  
```
